collection/rancher/v2.x/logs-collector/rancher2_logs_collector.sh

```diff
@@ -63,35 +63,47 @@ sherlock() {
     else
       echo -e "\n$(timestamp): couldn't detect OS"
   fi
-  if [ -n "${RUNTIME_FLAG}" ]
+  if [ -n "${DISTRO_FLAG}" ]
     then
-      techo "Setting container runtime as ${RUNTIME_FLAG}"
-      RUNTIME="${RUNTIME_FLAG}"
+      techo "Setting k8s distro as ${DISTRO_FLAG}"
+      DISTRO="${DISTRO_FLAG}"
     else
-      echo -n "$(timestamp): Detecting container runtime... "
+      echo -n "$(timestamp): Detecting k8s distribution... "
       if $(command -v docker >/dev/null 2>&1)
         then
           if $(docker ps >/dev/null 2>&1)
             then
-              RUNTIME=docker
-              echo "docker"
+              DISTRO=rke
+              echo "rke"
             else
-              FOUND="docker "
+              FOUND="rke "
           fi
       fi
       if $(command -v k3s >/dev/null 2>&1)
         then
           if $(k3s crictl ps >/dev/null 2>&1)
             then
-              RUNTIME=k3s
+              DISTRO=k3s
               echo "k3s"
             else
               FOUND+="k3s"
           fi
       fi
-      if [ -z $RUNTIME ]
+      if $(command -v rke2 >/dev/null 2>&1)
         then
-          echo -e "\n$(timestamp): couldn't detect container runtime"
+          RKE2_BIN="/var/lib/rancher/rke2/bin"
+          CRI_CONFIG_FILE="/var/lib/rancher/rke2/agent/etc/crictl.yaml"
+          if $(${RKE2_BIN}/crictl ps >/dev/null 2>&1)
```

Comment:
Tested on RKE2 and this check seems to fail and not detect the cluster is rke2. Ran the crictl ps command manually and get this:
```
root@ip-10-0-7-10:~# /var/lib/rancher/rke2/bin/crictl ps
WARN[0000] runtime connect using default endpoints: [unix:///var/run/dockershim.sock unix:///run/containerd/containerd.sock unix:///run/crio/crio.sock]. As the default settings are now deprecated, you should set the endpoint instead.
ERRO[0002] connect endpoint 'unix:///var/run/dockershim.sock', make sure you are running as root and the endpoint has been started: context deadline exceeded
WARN[0002] image connect using default endpoints: [unix:///var/run/dockershim.sock unix:///run/containerd/containerd.sock unix:///run/crio/crio.sock]. As the default settings are now deprecated, you should set the endpoint instead.
ERRO[0004] connect endpoint 'unix:///var/run/dockershim.sock', make sure you are running as root and the endpoint has been started: context deadline exceeded
FATA[0004] listing containers: rpc error: code = Unimplemented desc = unknown service runtime.v1alpha2.RuntimeService
```
This VM does have docker installed already, so not sure if that's interfering.

Reply:
Thanks, looks like `CRI_CONFIG_FILE` var needs to be inherited by subprocesses, I already had this exported on my nodes, added this to the script